site/_guides/gatekeeper.md

```diff
@@ -0,0 +1,406 @@
+```
+
+This creates a `projectcontour` namespace and sets up Contour as a deployment and Envoy as a daemonset, with communication between them secured by mutual TLS.
+
+Check the status of the Contour pods with this command:
+
+```bash
+$ kubectl -n projectcontour get pods -l app=contour
+NAME                           READY   STATUS      RESTARTS   AGE
+contour-8596d6dbd7-9nrg2       1/1     Running     0          32m
+contour-8596d6dbd7-mmtc8       1/1     Running     0          32m
+```
+
+If installation was successful, all pods should reach `Running` status shortly.
+
+## Deploy Gatekeeper
+
+The following instructions are summarized from the [Gatekeeper documentation](https://github.com/open-policy-agent/gatekeeper#installation-instructions).
+
+Run:
+
+```bash
+$ kubectl apply -f https://raw.githubusercontent.com/open-policy-agent/gatekeeper/master/deploy/gatekeeper.yaml
+```
+
+This creates a `gatekeeper-system` namespace and sets up the Gatekeeper controller manager and audit deployments using the latest Gatekeeper release.
+
+Check the status of the Gatekeeper pods with this command:
+
+```bash
+$ kubectl -n gatekeeper-system get pods
+NAME                                             READY   STATUS    RESTARTS   AGE
+gatekeeper-audit-67dfc46db6-kjcmc                1/1     Running   0          40m
+gatekeeper-controller-manager-7cbc758844-64hhn   1/1     Running   0          40m
+gatekeeper-controller-manager-7cbc758844-c4dkd   1/1     Running   0          40m
+gatekeeper-controller-manager-7cbc758844-xv9jn   1/1     Running   0          40m
+```
+
+If installation was successful, all pods should reach `Running` status shortly.
+
+## Configure Gatekeeper
+
+### Background
+
+Gatekeeper uses the [OPA Constraint Framework](https://github.com/open-policy-agent/frameworks/tree/master/constraint) to define and enforce policies.
+This framework has two key types: `ConstraintTemplate` and `Constraint`.
+A `ConstraintTemplate` defines a reusable OPA policy, along with the parameters that can be passed to it when it is instantiated.
+When a `ConstraintTemplate` is created, Gatekeeper automatically creates a custom resource definition (CRD) to represent it in the cluster.
+
+A `Constraint` is an instantiation of a `ConstraintTemplate`, which tells Gatekeeper to apply it to specific Kubernetes resource types (e.g. `HTTPProxy`) and provides any relevant parameter values.
+A `Constraint` is defined as an instance of the CRD representing the associated `ConstraintTemplate`.
+
+We'll now look at some examples to make these concepts concrete.
+
+### Configure resource caching
+
+First, Gatekeeper needs to be configured to store all `HTTPProxy` resources in its internal cache, so that existing `HTTPProxy` resources can be referenced within constraint template policies.
+This is essential for being able to define constraints that look across all `HTTPProxies` -- for example, to verify FQDN uniqueness.
```

Comment:
This is really interesting! I had assumed that Gatekeeper could only see the resource being admitted, plus some contextual data needed for ACLs. This idea of caching resources opens up a whole world of possibilities.